util/include/cppmicroservices/util/BundleElfFile.h

```diff
@@ -222,21 +228,33 @@ class BundleElfFile
       return iter->second;
     }
 
-    Shdr* strTblHdr = m_SectionHeaders + strTblHdrIdx;
+    const Shdr* const strTblHdr = m_SectionHeaders.get() + strTblHdrIdx;
+
+    if (strTblHdr && 0 > strTblHdr->sh_size) {
+      return nullptr;
+    }
 
     fs.seekg(strTblHdr->sh_offset);
-    char* strTbl = new char[static_cast<std::size_t>(strTblHdr->sh_size)];
-    fs.read(strTbl, strTblHdr->sh_size);
-    m_StrTblIndexToStrArray.insert(std::make_pair(strTblHdrIdx, strTbl));
+    char* strTbl = nullptr;
+    try {
+        strTbl = new char[static_cast<std::size_t>(strTblHdr->sh_size)];
```

Comment:
probably unrelated to this change but this whole file looks like a mix of old and new programming styles. 
explicit new/delete calls, malloc/free calls, functions returning C-style strings etc. Could be refactored to be more expressive and readable.

Reply:
I tried using a ```std::string``` instead of a ```char*``` in ```m_StrTblIndexToStrArray```, the problem is that the calling code doesn't want a string, it wants a pointer. I can try to make the ```char*``` a ```std::unique_ptr<char[]>``` instead.